### ISSUE TITLE
`operation` index allows `contracts` mapping to be empty

### ISSUE THREAD

**Issue by veqtor** [dipdup-io/dipdup#544]
**Steps to reproduce:**
Create a dipdup project, name it whatever, use this dipdup.yml:

```
spec_version: 1.2
package: fxhashbug

contracts:
  #FxHash
  FxHashModerationUser:
    address: KT1Wn2kkKmdbyLWBiLXWCkE7fKj1LsLKar2A
    typename: fxhash_moderation_user
  FxHashUserRegistry:
    address: KT1Ezht4PDKZri7aVppVGT4Jkw39sesaFnww
    typename: fxhash_user_registry

database:
  kind: sqlite
  path: fxhashbug.sqlite3

datasources:
  tzkt_mainnet:
    kind: tzkt
    url: https://api.tzkt.io

templates:
  fxhash:
    kind: operation
    datasource: <datasource>
    types:
      - transaction
    handlers:
      - callback: fxhash.admin.op.on_moderate
        pattern:
          - type: transaction
            destination: <fxhash_moderation_user>
            entrypoint: moderate
      - callback: fxhash.identity.op.on_update_profile
        pattern:
          - type: transaction
            destination: <fxhash_user_registry>
            entrypoint: update_profile


indexes:
  fxhashy:
    template: fxhash
    values:
      datasource: tzkt_mainnet
      fxhash_moderation_user: FxHashModerationUser
      fxhash_user_registry: FxHashUserRegistry
```

**What did you expect to happen:**

handlers are being called, it doesn't crash

**What actually happened:**
with `dipdup run` it crashes, see log:

[crash.log](https://github.com/dipdup-io/dipdup/files/10082980/crash.log)

when running in docker, handlers aren't called

### Environment

* **Python version:** 3.10
* **DipDup version or commit hash:**  6.3.0
* **Database engine version:**
* **Hasura version:**
* **Are you in docker?**
* **Can you reproduce this issue on `master`?**

config:

[config.txt](https://github.com/dipdup-io/dipdup/files/10082999/config.txt)

**Comment by droserasprout** [comment]
Hi @veqtor! Your bug reports are great, as always.

The reason for a crash is empty `contracts` mapping. To fix your template add the following lines:

```yaml
    contracts:
      - <fxhash_moderation_user>
      - <fxhash_user_registry>
```

Options for DipDup when the `operation` index has no contracts:

- Try to guess used contracts by handler patterns as it does for the `big_map` index (feels risky)
- Crash with `ConfigurationError`

**Comment by droserasprout** [comment]
Let's keep this one open; `RuntimeError` is not an expected behavior.

**Comment by droserasprout** [comment]
No `contracts` section is okay in some cases (originations). To not overcomplicate things it's better to crash with `ConfigurationError`.

Fixed in #543; lazy to push another branch.
